envoy/common/exception.h

```diff
@@ -57,15 +59,21 @@ class EnvoyException : public std::runtime_error {
 // the macros above.
 #define THROW_IF_STATUS_NOT_OK(variable, throw_action) THROW_IF_NOT_OK_REF(variable.status());
 
-#define RETURN_IF_STATUS_NOT_OK(variable)                                                          \
-  if (!variable.status().ok()) {                                                                   \
-    return variable.status();                                                                      \
-  }
+#define RETURN_IF_STATUS_NOT_OK(expr)                                                              \
+  do {                                                                                             \
```

Comment:
why not just have a local variable
absl::Status unusual_name_status = expr;
and avoid the wrapper class?
/wait-any
cc @toddmgreer for better hackarounds

Reply:
I assume the goal is for `RETURN_IF_NOT_OK(expr)` to only evaluate `expr` once, and that `expr` should be any expression that evaluates to an absl::Status (or something you can call `.ok()` on and return).

Seems like the following would work, but I could be overlooking something. It's been a while since I've practiced the dark arts of macrology.

#define RETURN_IF_NOT_OK(expr) \
  do {\
    /* Only evaluate `expr` once, and use temporary reference lifetime extension to refer to it twice. */ \
     const auto& status = expr;\
     if (!status.ok()) {\
       return status;\
       }\
  } while (false)
  
  Untested.